Test suite uncovers issues with latest numpy version

Hi,
Debian just upgraded to latest numpy which resulted in a [Bug report](https://bugs.debian.org/917693) against the python-skbio package.  The log says something like:

    ...
    ======================================================================
    ERROR: test_values_property (skbio.sequence.tests.test_sequence.TestSequence)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.7_skbio/build/skbio/sequence/tests/test_sequence.py", line 81, in setUp
        self.lowercase_seq = Sequence('AAAAaaaa', lowercase='key')
      File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.7_skbio/build/skbio/sequence/_sequence.py", line 647, in __init__
        self._convert_to_uppercase(lowercase_mask)
      File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.7_skbio/build/skbio/sequence/_sequence.py", line 676, in _convert_to_uppercase
        with self._byte_ownership():
      File "/usr/lib/python3.7/contextlib.py", line 112, in __enter__
        return next(self.gen)
      File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.7_skbio/build/skbio/sequence/_sequence.py", line 2191, in _byte_ownership
        self._bytes.flags.writeable = True
    ValueError: cannot set WRITEABLE flag to True of this array
  
    ======================================================================
    FAIL: test_init_lowercase_no_validation (skbio.sequence.tests.test_grammared_sequence.TestGrammaredSequence)
    ----------------------------------------------------------------------
    ValueError: cannot set WRITEABLE flag to True of this array
  
    During handling of the above exception, another exception occurred:
  
    Traceback (most recent call last):
      File "/<<PKGBUILDDIR>>/.pybuild/cpython3_3.7_skbio/build/skbio/sequence/tests/test_grammared_sequence.py", line 238, in test_init_lowercase_no_validation
        ExampleGrammaredSequence(s, lowercase=True)
    AssertionError: "character.*'R'" does not match "cannot set WRITEABLE flag to True of this array"
  
    ----------------------------------------------------------------------
    Ran 2340 tests in 42.261s
  
    FAILED (SKIP=30, errors=193, failures=1)

The numpy version that was used is 1.16.0~rc1.

Kind regards, Andreas.

--- a/skbio/sequence/_sequence.py
+++ b/skbio/sequence/_sequence.py
@@ -630,7 +630,7 @@ fuzzy=[(True, False)], metadata={'gene': 'foo'})
                                 type(sequence).__name__)
 
             sequence = s
-            self._owns_bytes = True
+            self._owns_bytes = False
 
             self._set_bytes(sequence)
 

